tests/dummy/app/components/memoized-computation.js

```diff
@@ -0,0 +1,40 @@
+import { withHooks, useMemo, useProperties } from "ember-hooks/mixins/ember-hooks";
+
+const MemoizedComponent = withHooks(() => {
+  const state = useProperties({
+    count: 0,
+    countDep: 0
+  });
+
+  const increment = () => {
+    state.count = state.count+ 1;
+  }
+
+  const incrementDep = () => {
+    state.countDep = state.countDep + 1;
+  }
+
+  // The callback function will only be ran when it's dependencies change.
+  // Without memoizing this or using a computed this would calculate on every render
+  // including when for example, the `count` property changed
+  const expensiveCount = useMemo(() => {
```

Comment:
I don't know how Ember's computeds work, but I know that useMemo is the simplest memoization case. I.E: Given the same set of dependencies return the previous result of the computation function,  otherwise return the result of the computation function. I.E. only one result/set of dependencies is cached.